0.data_analysis/construct_phenotypic_expression_plate_4_fs_data/nbconverted/cross_corr_all_cellprofiler.py

```diff
@@ -0,0 +1,349 @@
+output_path.mkdir(parents=True, exist_ok=True)
+
+
+# ## Load the data
+
+# In[5]:
+
+
+platedf = pd.read_parquet(path)
+
+
+# ## Specify data columns
+
+# In[6]:
+
+
+# The Column that corresponds to the concentration
+conc_col = "Metadata_Concentration"
+
+# The Column that corresponds to the construct
+construct_col = "Metadata_siRNA"
+
+# Columns that do not contain Metadata
+not_meta_cols = [col for col in platedf.columns if "Metadata" not in col]
+
+
+# ## Get cells that only contains NF1 siRNA constructs
+
+# In[7]:
+
+
+platedf[construct_col].fillna(value="None", inplace=True)
+
+platefilt = platedf[construct_col].str.contains("NF1")
+
+platedf = platedf.loc[platefilt]
+
+
+# ## Column for filtering correlation data
+
+# In[8]:
+
+
+platedf["Metadata_group"] = platedf.apply(
+    lambda row: f"{row['Metadata_Well']}_{row[conc_col]}_{row[construct_col]}", axis=1
+)
+
+
+# ## Find within well correlations
+# This will find the correlation values for each consctruct and concentration of siRNA
+
+# In[9]:
+
+
+inwell = []
+inwell = defaultdict(list)
+
+# Duplicates do not have to be filtered, since correlations across plates are not considered
+for group in platedf["Metadata_group"].unique():
+
+    # Find all cells that correspond to the specific (well, siRNA concentration, siRNA construct) group
+    welldf = platedf.loc[(platedf["Metadata_group"] == group)]
+
+    dfrow = welldf.iloc[0]
+
+    # The concentration of the siRNA construct for this group
+    conc = dfrow[conc_col]
+
+    # The construct of the siRNA construct for this group
+    construct = dfrow[construct_col]
+
+    # The well dataframe for this group
+    welldf = platedf.loc[platedf["Metadata_group"] == group]
+
+    # If there are no samples for a particular group, ignore it
+    if len(welldf) == 0:
+        continue
+
+    # Get the well dataframe that does not have metadata columns to compute correlations
+    welldf = welldf[not_meta_cols]
+
+    # Get a correlation matrix between each cell pair and covert it to a numpy array
+    corr_mat = welldf.T.corr().values
+
+    # Get the values that correspond to the lower triangle excluding the main diagonal
+    idx = np.tril_indices(corr_mat.shape[0], k=-1)
+    corr_vec = corr_mat[idx]
+
+    # Append correlations in a list (order doesn't matter in this case)
+    inwell[f"{construct}_{conc}"].append(corr_vec)
+
+
+# In[10]:
+
+
+# Concatenate all correlations calculated within each well
+inwell = {
+    sirna_group: np.concatenate(corr_vecs) for sirna_group, corr_vecs in inwell.items()
+}
+
+
+# ## Compute entire correlation matrix
+
+# In[11]:
+
+
+# Remove metadata columns and Compute all correlations for all cells
+allcorr = platedf[not_meta_cols].T.corr()
+
+# Set all of the correlation coeffients between the same cells as nans
+allcorr.values[np.diag_indices_from(allcorr)] = np.nan
+
+
+# ## Function for same and different construct correlations
+
+# In[12]:
+
+
+def get_correlations(same_construct):
```

Comment:
There is definitely a benefit to doing this, but I want other things to take precedence at the beginning. I guess I could include it in another file, but it would only be one function in the file. If I end up including more functions I may consider this.